TauAnalysis/MCEmbeddingTools/plugins/CollectionMerger.cc

```diff
@@ -71,21 +71,21 @@ void  CollectionMerger<T1,T2>::fill_output_obj_tracker(std::unique_ptr<MergeColl
 template <typename T1, typename T2>
 void  CollectionMerger<T1,T2>::fill_output_obj_calo(std::unique_ptr<MergeCollection > & output, std::vector<edm::Handle<MergeCollection> > &inputCollections)
 {
- std::map<uint32_t, BaseHit >   output_map;  
- // First merge the two collections again
- for (auto inputCollection : inputCollections){
-  for ( typename MergeCollection::const_iterator recHit = inputCollection->begin(); recHit!= inputCollection->end(); ++recHit ) {
-    DetId detIdObject( recHit->detid().rawId() );
-    T2 *akt_calo_obj = &output_map[detIdObject.rawId()];
-    float new_energy = akt_calo_obj->energy() + recHit->energy();
-    T2 newRecHit(*recHit);
-    newRecHit.setEnergy(new_energy);
-    *akt_calo_obj = newRecHit;
-  } 
+  std::map<uint32_t, BaseHit >   output_map;
+  // First merge the two collections again
+  for (auto inputCollection : inputCollections){
```

Comment:
@davidlange6 yes thanks, the inputCollections are only read. I will add this commit to the PR.